src/Collapse/utilities/distance_matrix_edge_collapse_rips_persistence.cpp

```diff
@@ -0,0 +1,177 @@
+
+using Field_Zp = Gudhi::persistent_cohomology::Field_Zp;
+using Persistent_cohomology = Gudhi::persistent_cohomology::Persistent_cohomology<Simplex_tree, Field_Zp>;
+using Distance_matrix = std::vector<std::vector<Filtration_value>>;
+
+void program_options(int argc, char* const argv[], double& min_persistence, double& end_thresold,
+                     int& dimension, int& dim_max, std::string& csv_matrix_file, std::string& filediag) {
+  namespace po = boost::program_options;
+  po::options_description visible("Allowed options", 100);
+      visible.add_options()
+        ("help,h", "produce help message")
+      	("min_persistence,m", po::value<double>(&min_persistence)->default_value(0.1),
+         "Minimum persistence interval length")
+        ("end_thresold,e", po::value<double>(&end_thresold)->default_value(1),
+         "Final threshold for rips complex.")
+        ("dimensions,D", po::value<int>(&dimension)->default_value(2),
+         "Dimension of the manifold.")
+        ("dim_max,k ", po::value<int>(&dim_max)->default_value(2),
+         "Maximum allowed dimension of the Rips complex.")
+        ("input_file_name,i", po::value<std::string>(&csv_matrix_file),
+         "The input file.")
+        ("filediag,o", po::value<std::string>(&filediag),
+         "The output file.");
+
+  po::options_description all;
+  all.add(visible);
+  po::variables_map vm;
+  po::store(po::command_line_parser(argc, argv).options(all).run(), vm);
+  po::notify(vm);
+  if (vm.count("help")) {
+    std::cout << std::endl;
+    std::cout << "Computes rips complexes of different threshold values, to 'end_thresold' from a n random uniform "
+                 "point_vector on a selected manifold, . \n";
+    std::cout << "Strongly collapses all the rips complexes and output the results in out_file. \n";
+    std::cout << "The experiments are repeted 'repete' num of times for each threshold value. \n";
+    std::cout << "type -m for manifold options, 's' for uni sphere, 'b' for unit ball, 'f' for file. \n";
+    std::cout << "type -i 'filename' for Input file option for exported point sample. \n";
+    std::cout << std::endl << std::endl;
+    std::cout << "Usage: " << argv[0] << " [options]" << std::endl << std::endl;
+    std::cout << visible << std::endl;
+    std::abort();
+  }
+}
+
+void program_options(int argc, char* argv[], std::string& csv_matrix_file, std::string& filediag,
+                     Filtration_value& threshold, int& dim_max, int& p, Filtration_value& min_persistence);
+
+int main(int argc, char* argv[]) {
+  std::string csv_matrix_file;
+  std::string filediag;
+  Filtration_value threshold;
+  int dim_max = 2;
+  int p;
+  Filtration_value min_persistence;
+
+  program_options(argc, argv, csv_matrix_file, filediag, threshold, dim_max, p, min_persistence);
+
+  Distance_matrix distances;
+
+  distances = Gudhi::read_lower_triangular_matrix_from_csv_file<Filtration_value>(csv_matrix_file);
```

Comment:
Could you put that on the same line as the declaration, to avoid copying?

Reply:
Fix on 4fdc721